packages/thirdweb/src/react/web/wallets/in-app/InAppWalletOTPLoginUI.tsx

```diff
@@ -21,60 +21,85 @@ type VerificationStatus =
   | "valid"
   | "idle"
   | "payment_required";
-type EmailStatus = "sending" | SendEmailOtpReturnType | "error";
-type ScreenToShow =
-  | "base"
-  | "create-password"
-  | "enter-password-or-recovery-code";
+type AccountStatus = "sending" | SendEmailOtpReturnType | "error";
+type ScreenToShow = "base" | "enter-password-or-recovery-code";
 
 /**
  * @internal
  */
 export function InAppWalletOTPLoginUI(props: {
+  userInfo: { email: string } | { phone: string };
   wallet: Wallet<"inApp">;
-  email: string;
   locale: InAppWalletLocale;
   done: () => void;
   goBack?: () => void;
 }) {
-  const { wallet, done, goBack } = props;
-  const email = props.email;
+  const { wallet, done, goBack, userInfo } = props;
   const { client, chain, connectModal } = useConnectUI();
   const isWideModal = connectModal.size === "wide";
   const locale = props.locale;
   const [otpInput, setOtpInput] = useState("");
   const [verifyStatus, setVerifyStatus] = useState<VerificationStatus>("idle");
-  const [emailStatus, setEmailStatus] = useState<EmailStatus>("sending");
+  const [accountStatus, setAccountStatus] = useState<AccountStatus>("sending");
 
   const [screen] = useState<ScreenToShow>("base");
 
-  const sendEmail = useCallback(async () => {
+  const sendEmailOrSms = useCallback(async () => {
     setOtpInput("");
     setVerifyStatus("idle");
-    setEmailStatus("sending");
+    setAccountStatus("sending");
 
     try {
-      const status = await preAuthenticate({
-        email,
-        strategy: "email",
-        client,
-      });
-      setEmailStatus(status);
+      if ("email" in userInfo) {
+        const status = await preAuthenticate({
+          email: userInfo.email,
+          strategy: "email",
+          client,
+        });
+        setAccountStatus(status);
+      } else {
+        const status = await preAuthenticate({
+          phoneNumber: userInfo.phone,
+          strategy: "phone",
+          client,
+        });
```

Comment:
can we make these explicit please (also below) - if email -> email, if phone -> phone -> if neither -> error

Reply:
done